pipenv/project.py

```diff
@@ -240,12 +240,42 @@ def requirements_exists(self) -> bool:
         return bool(self.requirements_location)
 
     def is_venv_in_project(self) -> bool:
-        if self.s.PIPENV_VENV_IN_PROJECT is False:
-            return False
-        return self.s.PIPENV_VENV_IN_PROJECT or (
-            self.project_directory
-            and os.path.isdir(os.path.join(self.project_directory, ".venv"))
-        )
+        """
+        There are three sources of truth to reconcile:
+
+        - PIPENV_VENV_IN_PROJECT environment variable
+        - venv_in_project directive in Pipfile
+        - an existing .venv directory
+
+        If the Pipfile and ENV var disagree, there are a few options:
```

Comment:
I'm personally torn on this question.

In my experience, I think it's nice for the Pipfile to take precedence, since the specific project is likely the most aware if there are localdev integrations that explicitly need the venv included or excluded (e.g. if you haven't implemented some symlink magic or volume overwriting, then you may actually very explicitly want to force the VENV *out* of the project directory or it might ruin `docker run` with volume mounts)

However, I realize this is very informed by my personal usage, so I'm obviously biased.